### PR TITLE
Documents atanh() function for Go Language

### PR DESCRIPTION
**Pull request by danlarrazabal**
### Description

This PR aims to document the atanh function for the Go programming language, and thus contributes a markdown file documenting said function.

This was performed as a task on the 'Learn Git & GitHub' course in Codecademy.

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: content/go/concepts/math-functions/terms/atanh/atanh.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+Tags:
+  - 'Math'
+  - 'Trigonometry'
+  - 'Functions'
+CatalogContent:
+  - 'learn-go'
+  - 'paths/computer-science'
+---
+
+The **`Atanh()`** function takes a float64 argument representing the angle whose inverse hyperbolic tangent value you want to find. It returns this inverse hyperbolic tangent as a float64 value that can range from negative infinity to positive infinity.
```
Comment (SSwiniarski) [inline content/go/concepts/math-functions/terms/atanh/atanh.md:16]:
```suggestion
The **`Atanh()`** function takes a float64 argument representing the angle whose inverse hyperbolic tangent value is to be returned. It returns this inverse hyperbolic tangent as a float64 value that can range from negative infinity to positive infinity.
```
We avoid first- and second-person pronouns in Docs.

File: content/go/concepts/math-functions/terms/atanh/atanh.md [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+
+# Syntax
+
+```pseudo
+result := math.Atanh(x)
+```
```
Comment (SSwiniarski) [inline content/go/concepts/math-functions/terms/atanh/atanh.md:22]: We generally have a sentence after the syntax code block explaining `x` and `result`.

File: content/go/concepts/math-functions/terms/atanh/atanh.md [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+func main() {
+	
+	fmt.Println(math.Atanh(0.5))
+    fmt.Println(math.Atanh(math.NaN()))
+
+} 
+```
+
+The output will be
+```shell
```
Comment (SSwiniarski) [inline content/go/concepts/math-functions/terms/atanh/atanh.md:43]:
````suggestion
The output will be:

```shell
````

File: content/go/concepts/math-functions/terms/atanh/atanh.md [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+0.5493061443340548
+NaN
+```
+
+# Source file
```
Comment (SSwiniarski) [inline content/go/concepts/math-functions/terms/atanh/atanh.md:48]: Remove this source file section. It isn't the kind of documentation we're doing in Docs.

**Review by SSwiniarski — CHANGES_REQUESTED** [review]
@danlarrazabal I did a first review and left a few comments.

File: content/go/concepts/math-functions/terms/atanh/atanh.md (hunk 2 above)
Comment (danlarrazabal) [reply content/go/concepts/math-functions/terms/atanh/atanh.md:22]: I have detailed the meanings of 'x' and 'result'

File: content/go/concepts/math-functions/terms/atanh/atanh.md (hunk 4 above)
Comment (danlarrazabal) [reply content/go/concepts/math-functions/terms/atanh/atanh.md]: I have eliminated the source file section

**Comment by danlarrazabal** [comment]
Thank you for your feedback @SSwiniarski 

I have gone ahead and implemented the requested changes.

**Review by SSwiniarski — APPROVED** [review]
@danlarrazabal Looks good for a second review.

File: content/go/concepts/math-functions/terms/atanh/atanh.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+
+# Syntax
+
+```pseudo
+result := math.Atanh(x)
+```
+
+Where `x` represents the angle whose inverse hyperbolic tangent value is to be returned.
+
+And where the result is the inverse hyperbolic tangent of `x`.
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/atanh/atanh.md:26]: Replace this line with "notable cases" that highlights results given specific values of x, e.g. result = -Inf when x = -1

File: content/go/concepts/math-functions/terms/atanh/atanh.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+# Example
+
+The following code calculates the arctangent and prints out the result:
+
+```go
+package main
+
+import "fmt"
+import "math"
```
Comment (caupolicandiaz) [inline content/go/concepts/math-functions/terms/atanh/atanh.md:36]:
```suggestion
import (
  "fmt"
  "math"
  )
```

Comment (caupolicandiaz) [reply content/go/concepts/math-functions/terms/atanh/atanh.md:36]: use 2 spaces for indentations in the code, not tabs

**Review by caupolicandiaz — CHANGES_REQUESTED** [review]
@danlarrazabal I've added a few comments. Please update the entry, and tag me when the changes are complete. Thanks

File: content/go/concepts/math-functions/terms/atanh/atanh.md (hunk 5 above)
Comment (danlarrazabal) [reply content/go/concepts/math-functions/terms/atanh/atanh.md:26]: Hello, @caupolicandiaz 

Thanks for the feedback.

Could you please help me understand a little better what you're looking for with regards to the "notable cases" section?

From your description, I find it very similar to the currently documented "Example" section, and I'm not sure to understand the differences.

If you could elaborate, I would sure appreciate it. It'd help me do a better job at complementing the documentation.

Once again, thank you for your feedback!

Comment (caupolicandiaz) [reply content/go/concepts/math-functions/terms/atanh/atanh.md:26]: Take a look at this example
https://www.codecademy.com/resources/docs/go/math-functions/cosh

Comment (danlarrazabal) [reply content/go/concepts/math-functions/terms/atanh/atanh.md]: Excuse the delay, @caupolicandiaz .

The notable cases of when 1 or -1 are passed to the function have been documented.

**Review by caupolicandiaz — APPROVED** [review]
@danlarrazabal @SSwiniarski LGTM

**Pull request merged**